jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt

```diff
@@ -0,0 +1,217 @@
+    // properties were set.
+    JitsiConfig.reloadNewConfig()
+
+    startIce4j()
+
+    val octoRelayService = org.jitsi.videobridge.octo.singleton().get()?.apply { start() }
+    val clientConnection = org.jitsi.videobridge.xmpp.singleton().get().apply { start() }
+    val statsMgr = org.jitsi.videobridge.stats.singleton().get()?.apply {
+        addStatistics(
+            VideobridgeStatistics(
+                // The Videobridge singleton
+                singleton().get(),
+                octoRelayService,
+                clientConnection
+            ),
+            StatsManager.config.interval.toMillis()
+        )
+        StatsManager.config.transportConfigs.forEach { transportConfig ->
+            addTransport(transportConfig.toStatsTransport(), transportConfig.interval.toMillis())
+        }
+        start()
+    }
+    org.jitsi.videobridge.health.singleton().get().start()
+
+    val publicHttpServer = setupPublicHttpServer()?.apply {
+        logger.info("Starting public http server")
+        start()
+    } ?: run {
+        logger.info("Not starting public http server")
+        null
+    }
+
+    val privateHttpServer = setupPrivateHttpServer()?.apply {
+        logger.info("Starting private http server")
+        start()
+    } ?: run {
+        logger.info("Not starting private http server")
+        null
+    }
+
+    org.jitsi.videobridge.shutdown.singleton().get().waitForShutdown()
+
+    logger.info("Bridge shutting down")
+    octoRelayService?.stop()
+    clientConnection.stop()
+    statsMgr?.stop()
+
+    try {
+        publicHttpServer?.stop()
+        privateHttpServer?.stop()
+    } catch (t: Throwable) {
+        t.printStackTrace()
+    }
+    org.jitsi.videobridge.health.singleton().get().stop()
+
+    // The Videobridge singleton
+    singleton().get().stop()
+
+    stopIce4j()
+
+    TaskPools.SCHEDULED_POOL.shutdownNow()
+    TaskPools.CPU_POOL.shutdownNow()
+    TaskPools.IO_POOL.shutdownNow()
+}
+
+private fun setupMetaconfigLogger() {
+    val configLogger = LoggerImpl("org.jitsi.config")
+    MetaconfigSettings.logger = object : MetaconfigLogger {
+        override fun warn(block: () -> String) = configLogger.warn(block)
+        override fun error(block: () -> String) = configLogger.error(block)
+        override fun debug(block: () -> String) = configLogger.debug(block)
+    }
+}
+
+private fun setupPublicHttpServer(): Server? {
+    val publicServerConfig = JettyBundleActivatorConfig(
+        "org.jitsi.videobridge.rest",
+        "videobridge.http-servers.public"
+    )
+    if (publicServerConfig.port == -1 && publicServerConfig.tlsPort == -1) {
+        return null
+    }
+    val publicServer = createServer(publicServerConfig)
+
+    val websocketService = ColibriWebSocketService(publicServerConfig.isTls)
+    org.jitsi.videobridge.websocket.singleton().setColibriWebSocketService(websocketService)
```

Comment:
You could import all these with shorter names, e.g. `websocketSingleton`

Reply:
Yeah, I can do it that way.